jax/numpy/lax_numpy.py

```diff
@@ -1240,6 +1240,26 @@ def select(condlist, choicelist, default=0):
   return output
 
 
+@_wraps(onp.bincount, lax_description="""\
+Jax adds the optional `length` parameter which specifies the output length, and
+defaults to ``x.max() + 1``. It must be specified for bincount to be compilable.
+""")
+def bincount(x, weights=None, minlength=0, *, length=None):
+  if length is None:
+    length = x.max() + 1
+  length = _max(length, minlength)
+  x = array(x)
+  if ndim(x) != 1:
+    raise ValueError("only 1-dimensional input supported.")
+  if weights is None:
+    weights = array(1, dtype='int32')
+  else:
+    if shape(x) != shape(weights):
+      raise ValueError("shape of weights must match shape of x.")
+    weights = array(weights)
+  return ops.index_add(zeros((length,), dtype(weights)), ops.index[x], weights)
```

Comment:
It's also worth noting the out-of-domain behavior here: updates out of the range `[0, length)` will be dropped entirely. Are we happy with this behavior? (Probably.) Perhaps document and perhaps test this behavior, given it will differ from classic NumPy?

An alternative would be to issue an error, but we don't in general have a mechanism for this yet.

Reply:
Ah I see you just did this. The same issue exists for values greater than `length` if provided.

Reply:
Noted this in the additional docs.